enhancements/storage/csi-driver-operator-merge.md

```diff
@@ -0,0 +1,379 @@
+QE. Before OCP feature freeze, we must decide which operator image will be shipped in the release. The other one will be
+removed from the release, i.e. from CI, from ART pipeline and from payload.
+
+Example (using BREW package names + AWS EBS CSI driver operator + 4.15):
+
+* In OCP 4.14, ART builds
+  only [`ose-aws-ebs-csi-driver-operator-container`](https://brewweb.engineering.redhat.com/brew/packageinfo?packageID=74505)
+  from https://github.com/openshift/aws-ebs-csi-driver-operator repository.
+
+* During OCP 4.15 development, we want ART to build both `ose-aws-ebs-csi-driver-operator-container` (as today)
+  and say `ose-aws-ebs-csi-driver-operator-v2-container` from https://github.com/openshift/csi-operator.
+  TODO: better name for the new images?
+
+  We will file these tickets / PRs ([source](https://art-dash.engineering.redhat.com/self-service/new-content)):
+  * Comet:
+    * New _build repository_ for `ose-aws-ebs-csi-driver-operator-v2-container`.
+    * New _delivery repository_ for `ose-aws-ebs-csi-driver-operator-v2-container`.
+    * Shall we follow the same brew / dist-git / image names as the old one, just add `-v2-`? Or shall we fix the inconsistencies (e.g. `azure-file-...` vs `ose-azure-disk-...`)? 
+  * openshift/release:
+    * build the image in CI + promote to 4.15.
+  * ART:
+    * Follow https://art-dash.engineering.redhat.com/self-service/new-content, using almost the same data as in the existing image (e.g. multiarch support)
+      * Exception: we will not perform threat model assessment - we're merging code from existing repos.
+    
+* Before OCP 4.15 feature freeze we will decide if the `-v2-` image is good enough and file tickets to
+  disable builds of either the old or `-v2-` image. Similarly, we will update `cluster-storage-operator` and its feature gate
+  to use only one of these images.
+  We will file these tickets + PRs ([source](https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#removing-a-component-from-the-openshift-release-payload)):
+  * PR to update `cluster-storage-operator` and its related-images, so only the "good" image is used.
+    * This can be merged before anything below.
+  * Jira for ART to stop building the "bad" image, https://issues.redhat.com/browse/ART-1443
+    * They will create a PR to update ocp-build-data.
+  * When removing the old image: PR to openshift/release to stop building + promoting the image in CI.
+  * On slack, coordinate with ART merging of all these PRs
+  
+* At 4.15 GA (and in the whole 4.15.z stream), ART will build and ship only the "good" image.
+
+* In 4.16, if the "good" image is the old one, we re-submit all tickets to enable building + shipping of the `-v2-` image
+  again and continue testing.
+    * Drawback: we will have to stop its development + testing until OCP 4.16 is branched, so we can keep 4.15 with only
+      a single image until GA.
+
+Drawbacks:
+
+* New images need to be built, i.e. lot of Comet requuests.
```

Comment:
Additional drawback: Right now we have several years of history in comet for the existing repos, we can see all the builds for all the releases in one place for each operator. With this approach we will have 2 comet repos for each operator, and they will each have their own cut-over time to the new repo.

For example we would need some way to track this as we roll out changes for each operator:
- AWS EBS: -container repo for < 4.15 builds and -v2-container repo for >= 4.15 builds
- Azure Disk: -container repo for < 4.16 builds and -v2-container repo for >= 4.16 builds
- GCP PD: -container repo for < 4.17 builds and -v2-container repo for >= 4.17 builds

It could be managed, but it's more complexity for us, QE, and ART team. I guess we can never get rid of the old container repos until we completely stop supporting 4.15 for example.

Reply:
Drawback added.